metaverse_modules/heal-effect/index.js

```diff
@@ -0,0 +1,426 @@
+        
+      });
+      //######################################################## initial instanced mesh #################################################################
+      let flashMesh=null;
+      const addInstancedMesh=()=>{
+          const geometry2 = new THREE.PlaneGeometry( 0.5, 0.5 );
+          const geometry =_getFlashGeometry(geometry2)
+          flashMesh = new THREE.InstancedMesh(
+              geometry,
+              flashMaterial,
+              flashParticleCount
+          );
+          
+          const idAttribute = flashMesh.geometry.getAttribute('id');
+          for (let i = 0; i < flashParticleCount; i++) {
+              idAttribute.setX(i,i);
+          }
+          idAttribute.needsUpdate = true;
+      }
+      addInstancedMesh();
+      const pixel = new THREE.Points(pixelGeometry, pixelMaterial);
+      group.add(pixel);
+
+      let storeMaterial = false;
+      let particleAlreadyInScene= false;
+      let circlePlay = false;
+      let materialStartTime = -1;
+      let healMaterial = [];
+      let currentModel = null;
+      let dir = new THREE.Vector3();
+      useFrame(({timestamp}) => {
+          //#################### store avatar material ############################
+          const _initHealMaterial = () => {
+            healMaterial = [];
+            localPlayer.avatar.object.scene.traverse(o => {
+              // console.log(o.name, o.isMesh);
+              if (o.isMesh) {
+                if(o.material.constructor.name=='Array'){
+                  if (o.material[0].constructor.name=='MToonMaterial') {
+                    healMaterial.push(o.material[0].uniforms.rimColor.value);
+                  }
+                  else{
+                    if(o.material[0].emissive!==null && o.material[0].emissive!==undefined){
+                      healMaterial.push(o.material[0].emissive);
+                    }  
+                  }
+                }
+              }
+            });
+          };
+
+          if(currentModel !== localPlayer.avatar.model.uuid){
+            _initHealMaterial();
+            currentModel = localPlayer.avatar.model.uuid
+          }
+          //######################### flash attribute ##############################
+          const swAttribute = flashMesh.geometry.getAttribute('sw');
+          const positionsAttribute = flashMesh.geometry.getAttribute('positions');
+          const scalesAttribute = flashMesh.geometry.getAttribute('scales');
+          const idAttribute = flashMesh.geometry.getAttribute('id');
+
+          //######################### pixel attribute ##############################
+          const pixelOpacityAttribute = pixel.geometry.getAttribute('opacity');
+          const pixelPositionAttribute = pixel.geometry.getAttribute('position');
+          const pixelScaleAttribute = pixel.geometry.getAttribute('scales');
+          
+          if(localPlayer.hasAction('cure')){
+              localPlayer.removeAction('cure')
```

Comment:
Probably should only remove the action when the cure is done.

Reply:
Also this should apply to any player with the action, not just local.

This heal app should not really be deciding things based on actions. Instead the character fx should detect the action and then control this app based on that.

Reply:
[move detect action to character-fx](https://github.com/webaverse/app/pull/2873/commits/beedaac98cc0c13ab5d25f5f5df4f0bb3ac2e868)

Reply:
[remove localPlayer](https://github.com/webaverse/app/pull/2873/commits/f4c495dbc96c6133e6158df73702a92dd13b67d7)